src/api/spec/models/obs_factory/distribution_spec.rb

```diff
@@ -243,5 +243,37 @@ def strategy_for(name)
     it { expect(distribution.rings_project_name).to eq('openSUSE:Factory:Rings') }
   end
 
-  # TODO: OpenQA => openqa_filter and openqa_jobs_for
+  describe '#openqa_filter' do
+    let(:staging_project_a) { create(:project, name: 'openSUSE:Factory:Staging:A') }
+    let(:staging_project) { ObsFactory::StagingProject.new(project: staging_project_a, distribution: distribution) }
+
+    it { expect(distribution.openqa_filter(staging_project)).to eq('match=Staging:A') }
+  end
+
+  describe '#openqa_jobs_for' do
+    before do
+      allow(ObsFactory::OpenqaJob).to receive(:find_all_by)
+    end
+
+    shared_examples 'calls find_all_by' do |version, expected_build|
+      it "performs find_all_by with #{version} version" do
+        allow(distribution).to receive("#{version}_version").and_return(expected_build)
+        distribution.openqa_jobs_for(version)
```

Comment:
I would move those two lines to a before block. Or at least the stub in line 260

Reply:
Even if each 'allow' is used only in one 'it' block?

Reply:
Well, I tried to put that 'allow' inside the 'before' block and it calls 'allow' 9 times. 3 times per 'it'.
 It is even worst, isn't it?

Reply:
You know it isn't :smile:

Reply:
I meant within the shared_example:-)

Like:
```ruby
shared_examples 'calls find_all_by' do |version, expected_build|
  before do
    ...
  end

  it { ... }
end


```

Anyway, this is just a [nitpick](https://github.com/openSUSE/open-build-service/blob/master/CONTRIBUTING.md#how-to-review-code-submissions). You don't have to do it that way. It's just that I find splitting test preparation from the actual test makes them more readable.

Reply:
Yes, I tried with 'before' block inside 'shared_examples'... Well, we think it's better to keep it that way.